File: src/SIL.XForge.Scripture/Models/Answer.cs
using System.Collections.Generic;

namespace SIL.XForge.Scripture.Models;

public class Answer : Comment
{
    public VerseRefData? VerseRef { get; set; }
    public string? ScriptureText { get; set; }
    public List<Like> Likes { get; set; } = [];
    public List<Comment> Comments { get; set; } = [];
    public string Status { get; set; } = AnswerStatus.None;
}

File: src/RealtimeServer/scriptureforge/models/answer.ts
import { Comment } from './comment';
import { DynamicValue } from './dynamic-value';
import { Like } from './like';
import { VerseRefData } from './verse-ref-data';

export enum AnswerStatus {
  None = '',
  Resolved = 'resolved',
  Exportable = 'export'
}

export interface Answer extends Comment, DynamicValue {
  verseRef?: VerseRefData;
  scriptureText?: string;
  selectionStartClipped?: boolean;
  selectionEndClipped?: boolean;
  likes: Like[];
  comments: Comment[];
  status?: AnswerStatus;
}
